assets/js/data/cart/actions.js

```diff
@@ -242,3 +242,66 @@ export function* changeCartItemQuantity( cartItemKey, quantity ) {
 
 	yield itemQuantityPending( cartItemKey, false );
 }
+
+/**
+ * Selects a shipping rate.
+ *
+ * @param {string} rateId the id of the rate being selected.
+ * @param {number} packageId the key of the packages that we will select within.
+ */
+export function* selectShippingRate( rateId, packageId = 0 ) {
+	try {
+		const result = yield apiFetch( {
+			path: `/wc/store/cart/select-shipping-rate/${ packageId }`,
+			method: 'POST',
+			data: {
+				rate_id: rateId,
+			},
+			cache: 'no-store',
+		} );
+
+		if ( result ) {
+			yield receiveCart( result );
+		}
+	} catch ( error ) {
+		yield receiveError( error );
+	}
+}
+
+/**
+ * Returns an action object used to track what shipping address are we updating to.
+ *
+ * @param {boolean} isResolving if we're loading shipping address or not.
+ * @return {Object} Object for action.
+ */
+export function receiveShippingAddress( isResolving ) {
+	return {
+		type: types.UPDATING_SHIPPING,
+		isResolving,
+	};
+}
```

Comment:
The name of the action doesn't match what it's doing (nor does the description for that matter).  Since this is merely for tracking when shipping rates async update is happening (because a new address is being submitted), let's be more precise in the action name, and description.

So something like `shippingRatesAreResolving` or `shippingAddressIsUpdating`.  Since the selector name is `areShippingRatesLoading` I think `shippingRatesAreResolving` would be the better fit here.